outwatch/src/main/scala/outwatch/dom/helpers/NativeHelpers.scala

```diff
@@ -1,19 +1,41 @@
 package outwatch.dom.helpers
 
+import org.scalajs.dom.Element
+import org.scalajs.dom.raw.CSSStyleDeclaration
+
 import scala.scalajs.js
 import scala.scalajs.js.annotation.JSBracketAccess
 
+object JSDefined {
+  // provides an extractor for js.UndefOr
+  // https://gitter.im/scala-js/scala-js?at=5c3e221135350772cf375515
+  def apply[A](a: A): js.UndefOr[A] = a
+  def unapply[A](a: js.UndefOr[A]): UnapplyResult[A] = new UnapplyResult(a)
+
+  final class UnapplyResult[+A](val self: js.UndefOr[A])
```

Comment:
for performance reasons we might want to drop this extra allocation :)

We could use `.fold` on the js.undefor instead.